hibernate-gorm/src/main/groovy/io/micronaut/configuration/hibernate/gorm/HibernateDatastoreFactory.groovy

```diff
@@ -63,30 +63,27 @@ class HibernateDatastoreFactory {
             new ConfigurableEventPublisherAdapter(applicationContext),
             classes
         )
-        for (o in datastore.getServices()) {
-            applicationContext.registerSingleton(o, false)
+        SessionFactory sessionFactory = datastore.sessionFactory
+        DataSource dataSource = ((HibernateConnectionSource) datastore.getConnectionSources().defaultConnectionSource).getDataSource()
+        PlatformTransactionManager platformTransactionManager = datastore.getTransactionManager()
+        applicationContext.registerSingleton(SessionFactory, sessionFactory)
+        applicationContext.registerSingleton(DataSource, dataSource)
+        applicationContext.registerSingleton(PlatformTransactionManager, platformTransactionManager, Qualifiers.byName("hibernate"))
+        applicationContext.registerSingleton(PlatformTransactionManager, platformTransactionManager, Qualifiers.byStereotype(Primary))
+
+        Collection<Collection<Service>> gormServices = datastore.getServices().split {
+            !it.class.isAnnotationPresent(grails.gorm.services.Service)
         }
-        for (o in datastore.getServices()) {
-            applicationContext.inject(o)
+        for(services in gormServices) {
+            for (o in services) {
+                applicationContext.registerSingleton(o, false)
+            }
+            for (o in services) {
+                applicationContext.inject(o)
+            }
```

Comment:
Divides the services by those that simply implement the Service interface (e.g. those already provided by GORM) and those which have it weaved in via the @Service annotation (those provided by the project), and then registers them in that order.